xdsl/dialects/omp.py

```diff
@@ -141,6 +149,34 @@ class VariableCaptureKind(StrEnum):
     VLAType = "VLAType"
 
 
+class LoopWrapper(OpTrait):
+    """
+    Check that the omp operation is a loop wrapper as defined upstream.
+
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/OpenMPDialect/#loop-associated-directives)
+    """
+
+    def verify(self, op: Operation) -> None:
+        if not op.has_trait(NoTerminator()):
+            raise VerifyException(
+                f"{op.name} is not a LoopWrapper: does not have NoTerminator trait"
+            )
+        if (num_regions := len(op.regions)) != 1:
+            raise VerifyException(
+                f"{op.name} is not a LoopWrapper: has {num_regions} region, expected 1"
+            )
+        if (num_ops := len(op.regions[0].block.ops)) != 1:
+            raise VerifyException(
+                f"{op.name} is not a LoopWrapper: has {num_ops} ops, expected 1"
+            )
+        assert (inner := op.regions[0].block.first_op), "Expected exactly 1 op"
+        if not (inner.has_trait(LoopWrapper()) or isa(inner, LoopNestOp)):
```

Comment:
```suggestion
        if not (inner.has_trait(LoopWrapper()) or isinstance(inner, LoopNestOp)):
```
`isinstance` is sufficient here.